bundles/org.openhab.binding.neohub/src/test/java/org/openhab/binding/neohub/test/NeoHubProtocolTests.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.neohub.test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.junit.jupiter.api.Test;
+import org.openhab.binding.neohub.internal.NeoHubBindingConstants;
+import org.openhab.binding.neohub.internal.NeoHubConfiguration;
+import org.openhab.binding.neohub.internal.NeoHubSocket;
+import org.openhab.binding.neohub.internal.NeoHubWebSocket;
+
+/**
+ * JUnit for testing WSS and TCP socket protocols.
+ *
+ * @author Andrew Fiddian-Green - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class NeoHubProtocolTests {
+
+    /**
+     * Test online communication. Requires an actual Neohub to be present on the LAN. Configuration parameters must be
+     * entered for the actual specific Neohub instance as follows:
+     *
+     * - HUB_IP_ADDRESS the dotted ip address of the hub
+     * - HUB_API_TOKEN the api access token for the hub
+     * - SOCKET_TIMEOUT the connection time out
+     * - RUN_WSS_TEST enable testing the WSS communication
+     * - RUN_TCP_TEST enable testing the TCP communication
+     *
+     * NOTE: only run these tests if a device is actually available
+     *
+     */
+    private static final String HUB_IP_ADDRESS = "192.168.1.xxx";
+    private static final String HUB_API_TOKEN = "12345678-1234-1234-1234-123456789ABC";
+    private static final int SOCKET_TIMEOUT = 5;
+    private static final boolean RUN_WSS_TEST = false;
+    private static final boolean RUN_TCP_TEST = false;
+
+    /**
+     * Use web socket to send a request, and check for a response.
+     */
+    @Test
+    void testWssConnection() {
+        if (RUN_WSS_TEST) {
+            if (!NeoHubJsonTests.VALID_IP_V4_ADDRESS.matcher(HUB_IP_ADDRESS).matches()) {
+                fail();
+            }
+
+            NeoHubConfiguration config = new NeoHubConfiguration();
+            config.hostName = HUB_IP_ADDRESS;
+            config.socketTimeout = SOCKET_TIMEOUT;
+            config.apiToken = HUB_API_TOKEN;
+
+            try {
+                NeoHubWebSocket socket = new NeoHubWebSocket(config);
+                String requestJson = NeoHubBindingConstants.CMD_CODE_FIRMWARE;
+                String responseJson = socket.sendMessage(requestJson);
+                assertNotEquals(0, responseJson.length());
+                socket.close();
+            } catch (Exception e) {
+                fail(e.getMessage());
```

Comment:
If you remove this catch, you can potentially have more information shown as failed test output if any exception is thrown (e.g. stack trace). Checked exceptions can be added to the method signature.

Reply:
I am not really sure what you mean by this statement. The test is to confirm that one can connect to a hub via a websocket, send a query, and return a result. So it is a test for one positive; and NOT a test for multiple potential negatives. The catch clause is basically only present to satisfy the compiler. If I would want to test for 'N' multiple potential negatives then I would need to write 'N@ test cases..

Reply:
It was a small tip to simplify your test - I'll post it as a new comment so you can see the diff.

Reply:
The proposed version is not only simpler, it also provides better output in case an exception will be thrown. You can resolve both comments if you prefer not to change anything.